plugins/csi/client.go

```diff
@@ -393,36 +401,40 @@ NEXT_CAP:
 		expectedMode := expected.GetAccessMode().GetMode()
 		capMode := cap.GetAccessMode().GetMode()
 
-		if expectedMode != capMode {
-			multierror.Append(&err,
-				fmt.Errorf("requested AccessMode %v, got %v", expectedMode, capMode))
-			continue NEXT_CAP
+		// The plugin may not validate AccessMode, in which case we'll
+		// get UNKNOWN as our response
+		if capMode != csipbv1.VolumeCapability_AccessMode_UNKNOWN {
+			if expectedMode != capMode {
+				multierror.Append(&err,
+					fmt.Errorf("requested access mode %v, got %v", expectedMode, capMode))
+				continue NEXT_CAP
+			}
 		}
 
-		// AccessType Block is an empty struct even if set, so the
-		// only way to test for it is to check that the AccessType
-		// isn't Mount.
-		expectedMount := expected.GetMount()
+		capBlock := cap.GetBlock()
 		capMount := cap.GetMount()
+		expectedBlock := expected.GetBlock()
+		expectedMount := expected.GetMount()
 
-		if expectedMount == nil {
-			if capMount == nil {
-				return nil
-			}
+		if capBlock != nil && expectedBlock == nil {
 			multierror.Append(&err, fmt.Errorf(
-				"requested AccessType Block but got AccessType Mount"))
+				"'block-device' access type was not requested but was validated by the controller"))
```

Comment:
Note to reviewers: I've changed the wording of these error messages so that they match what the operator has submitted in the volume spec or jobspec, rather than matching the CSI internal nomenclature.